webapp/src/components/data/dataset_input/FileSelection.vue

```diff
@@ -90,76 +94,161 @@
+          <div>  
+            <CustomButton @click="clearFiles">
+              clear file{{ isMultiple ? 's' : '' }}
+            </CustomButton>
           </div>
-          <CustomButton @click="clearFiles">
-            clear file{{ isMultiple ? 's' : '' }}
-          </CustomButton>
         </div>
       </section>
     </article>
   </div>
 </template>
 
 <script lang="ts" setup>
-import { ref, withDefaults } from 'vue'
+import { ref, withDefaults, computed } from 'vue'
+import * as d3 from 'd3'
 
 import CustomButton from '@/components/simple/CustomButton.vue'
+import type { Task } from '@epfml/discojs'
+import { data } from '@epfml/discojs'
+import { useToaster } from '@/composables/toaster'
+
+const toaster = useToaster()
+
+
+type CSV = { filename: string, label: string }[] | undefined
 
-interface Emits {
-  (e: 'input', files: FileList): void
-  (e: 'clear'): void
+interface Emits { 
+  (e: 'csv', csvRows: CSV): void
 }
 
 interface Props {
+  task: Task
+  datasetBuilder: data.DatasetBuilder<File>
+  csvRows: CSV,
+  label?: string,
   isDirectory?: boolean,
   acceptFiles?: string[],
   isMultiple?: boolean,
-  infoText?: boolean
+  infoText?: boolean,
+  expectCsvMapping?: boolean
 }
+const props = withDefaults(defineProps<Props>(), {
+  isDirectory: false,
+  isMultiple: true,
+  acceptFiles: () => ['*'],
+  infoText: false,
+  label: undefined,
+  expectCsvMapping:false
+})
+
 const emit = defineEmits<Emits>()
 const selectedFiles = ref<FileList>()
 
 const uploadFile = ref<HTMLInputElement>()
 const uploadDirectory = ref<HTMLInputElement>()
 
-withDefaults(defineProps<Props>(), {
-  isDirectory: false,
-  isMultiple: true,
-  acceptFiles: () => ['*'],
-  infoText: false
+const hideConnectField = ref(false)
+
+const fileType = computed(() => {
+  if (props.expectCsvMapping === true || props.task.trainingInformation.dataType === 'tabular') {
+    return 'CSV'
+  } 
+  return props.task.trainingInformation.dataType === 'image' ? 'image' : 'file'
 })
+const requireLabels = computed(
+  () => props.task.trainingInformation.LABEL_LIST !== undefined
+)
 
-const clearFiles = () => {
-  emit('clear')
-  selectedFiles.value = undefined
-}
 const submitFiles = () => {
   if (uploadFile.value === undefined) return
   const files = uploadFile.value.files
   if (files === null) return
-
-  emit('input', files)
-  selectedFiles.value = files
+  addFiles(files)
 }
 const submitDirectory = () => {
   if (uploadDirectory.value === undefined) return
   const files = uploadDirectory.value.files
   if (files === null) return
-
-  emit('input', files)
-  selectedFiles.value = files
+  addFiles(files)
 }
+
 const dragFiles = (e: DragEvent) => {
   if (e.dataTransfer === null) return
-
   e.dataTransfer.dropEffect = 'copy'
-  emit('input', e.dataTransfer.files)
-  selectedFiles.value = e.dataTransfer.files
+  const files = e.dataTransfer.files
+  addFiles(files)
 }
+
+const addFiles = (files: FileList) => {
+  if (props.expectCsvMapping === true) {
```

Comment:
it's always def and boolean

```suggestion
  if (props.expectCsvMapping) {
```